include/small/detail/container/associative_vector.hpp

```diff
@@ -805,12 +786,33 @@ namespace small::detail {
         /// \brief Erase element at a position in small map
         size_type
         erase(const key_type &k) {
-            auto it = find(k);
-            if (it != end()) {
-                erase(it);
-                return 1;
+            return erase<decltype(k)>(k);
```

Comment:
Isn't `decltype(k)` always the same? `const key_type&`?

Reply:
It is.

Maybe this is just a matter of taste, I think this is better than writing out the type twice.